storage/remote/wal_watcher.go

```diff
@@ -0,0 +1,278 @@
+// Copyright 2018 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remote
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"regexp"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/tsdb"
+	"github.com/prometheus/tsdb/wal"
+	fsnotify "gopkg.in/fsnotify/fsnotify.v1"
+)
+
+// WALWatcher watches the TSDB WAL for a given StorageClient (remote write config).
+type WALWatcher struct {
+	storage *Storage
```

Comment:
We're going to go with each queue tailing the WAL because:
- when we have 0 remote writes, we don't want to tail
- when we have a single remote endpoint down, we don't to to block all

Reply:
Also, lets consider decoupling the WALWatcher from the Queue using an interface, as I suspect other people will use this.